extensions-contrib/confluent-extensions/src/main/java/io/confluent/druid/transform/TenantUtils.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ */
+
+package io.confluent.druid.transform;
+
+import javax.annotation.Nullable;
+
+public class TenantUtils
+{
+  private static final char DELIMITER = '_';
```

Comment:
I wonder if we can make this generic by 
1. Making the delimiter configurable?
2. Merge the conditions on lines 16 and 21 and make it configurable to pick either the left token or right token?

If we do this, we could get a "split string" transform that splits the string based on a delimiter and lets you pick either the left or the right token.

Reply:
we could, but I'd rather make those specific for their purpose, since those transforms are not really behaving like typical ones, i.e. they preserve existing values for the target field if present rather than over-writing it.

Reply:
I think that a generic "split string" transform could be useful for other cases with different delimiters(especially when we start ingesting events in druid). And we can make the overwrite behavior configurable too. 

But I agree with you that we can generalize this later. We can wait for the use cases to show up and then decide on this in the future.

Reply:
I am able to use "nvl" function to get tenant not overwritten if tenant is present.
Would this make "strsplit" suggested by @arrawatia  more in line with what we want to achieve? 
```"transforms": [
        {
          "type": "expression",
          "name": "tenant",
          "expression": "nvl(\"tenant\",regexp_extract(\"topic\", '(^[^_]+)',1))"
        },
        {
          "type": "expression",
          "name": "tenant_topic",
          "expression": "regexp_extract(\"topic\", '^[^_]*_(.*)', 1)"
        }
      ]```

Reply:
no, `nvl` doesnt' achieve the same thing. The goal here is to avoid overriding the value if the _target_ field is already not null.

Reply:
hmm, I tested it, nvl() works as expected. https://druid.apache.org/docs/latest/misc/math-expr.html
```nvl(expr,expr-for-null) returns 'expr-for-null' if 'expr' is null (or empty string for string type)```

Reply:
ah good call, might be good to compare performance of expressions vs. the custom transforms